### PR TITLE
Fix/94 url params

### PR DESCRIPTION
**Pull request by rybalkinsd**

### CODE REVIEW

File: src/main/kotlin/io/github/rybalkinsd/kohttp/ext/HttpContextExt.kt [hunk 1]
```diff
@@ -14,10 +14,21 @@ fun HttpContext.url(url: URL) {
...
     host = url.host ?: throw IllegalArgumentException("unexpected host: $host")
 
-    if (url.port != -1) {
+    if (url.port != -1)
         port = url.port
-    }
+
     path = url.path
+
+    if (url.query?.isNotBlank() == true) {
```
Comment (gokulchandra) [inline src/main/kotlin/io/github/rybalkinsd/kohttp/ext/HttpContextExt.kt]: The condition can be simplified. `if (url.query?.isNotBlank())`

Comment (rybalkinsd) [reply src/main/kotlin/io/github/rybalkinsd/kohttp/ext/HttpContextExt.kt]: Unfortunately, `url.query?.isNotBlank()` produces `Boolean?` which can not be used as is inside `if` statement

File: src/main/kotlin/io/github/rybalkinsd/kohttp/ext/HttpContextExt.kt [hunk 2]
```diff
@@ -14,10 +14,21 @@ fun HttpContext.url(url: URL) {
...
+        param {
+            url.query.split("&")
+                .map { it.split("=", limit = 2) }
+                .groupBy({ it[0] }, { it.getOrNull(1) })
```
Comment (rybalkinsd) [inline src/main/kotlin/io/github/rybalkinsd/kohttp/ext/HttpContextExt.kt]: probably, grouping op here is not necessary, b/c `ParamContext` applies add in a special way

File: src/test/kotlin/io/github/rybalkinsd/kohttp/Utils.kt [hunk 3]
```diff
@@ -1,13 +1,44 @@
...
 import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.node.ArrayNode
 import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+/**
+ * todo: Probably it would be better to provide extension function to test Json
+ * @author sergey
+ */
+fun assertContainsExactly(expected: Map<String, Any?>, actual: JsonNode) {
```
Comment (rybalkinsd) [inline src/test/kotlin/io/github/rybalkinsd/kohttp/Utils.kt]: @gokulchandra , Looks our codebase becoming bigger and bigger. 
I had to split `assertResponses` function into two. 
1st - to check we have `atLeast` some "nodes" inside
2nd - to check identity of nodes

How do you feel about making this two more elegant and kotlin-way?
Currently we do not check list content in `atLeast`:  `list(a, b)` contains `list(a)` probably should result to `true`
Also current code could be simplified, I believe

**Comment by rybalkinsd** [comment]
this PR should fix https://github.com/rybalkinsd/kohttp/issues/118

File: src/test/kotlin/io/github/rybalkinsd/kohttp/Utils.kt (hunk 3 above)
Comment (gokulchandra) [reply src/test/kotlin/io/github/rybalkinsd/kohttp/Utils.kt]: We could compare number of keys in `expected` and `actual`. And fail if they are different. Something like:
```
fun assertContainsExactly(expected: Map<String, Any?>, actual: JsonNode) {
	    assertContainsAtLeast(expected, actual)
		assertEquals(actual.fieldNames().length, expected.fieldNames.length())
}
```

File: src/test/kotlin/io/github/rybalkinsd/kohttp/Utils.kt [hunk 4]
```diff
@@ -1,13 +1,44 @@
...
-        assertEquals(u, actual[t]?.asText())
+fun assertContainsAtLeast(expected: Map<String, Any?>, actual: JsonNode) {
+    expected.forEach { (k, v) ->
+        when (v) {
+            is List<*> -> {
+                val arrayNode = actual[k] as? ArrayNode ?: throw Exception("$k type is not Array, as expected")
+
+                assertEquals(v.size, arrayNode.size())
+
+                arrayNode.forEachIndexed { i, element ->
```
Comment (gokulchandra) [inline src/test/kotlin/io/github/rybalkinsd/kohttp/Utils.kt]: Can we leverage `.containsAll` here to check if `actual` has all members of `expected`?

**Pull request merged**